### ISSUE TITLE
Flag import support for Click API

### ISSUE THREAD

**Issue by gar1t** [guildai/guildai#48]
[Click] is a handy way to add CLI support to a script (see below). It'd be nice if Guild supported Click the way it supports argparse.

### Upgrading from a Notebook style script to a CLI

In keeping with Notebooks style development, here's a script that hard codes everything:

```python
msg = "yop"
print(msg)
```

This is trivial but its behavior can't be modified without changing the source code. While Guild can run this script by modifying `msg` when it loads the module (AST magic) it'd be nice to have a reusable script that can be run and tested independent of Guild.

Here's how you'd modify the script to support a CLI using `argparse`:

``` python
import argparse

p = argparse.ArgumentParser()
p.add_argument("msg", default="yop")
args = p.parse_args()

print(args.msg)
```

That's pretty easy! It's also the recommended approach for building scripts for Guild because it supports an explicit command line interface.

Here's the script using Click:

``` python
import click

@click.command()
@click.argument("msg", default="yop")
def hello(msg):
    print(msg)

hello()
```

There is an additional line here, but the Click convention of decorating functions makes it easy to build CLIs with more capability.

**Comment by samedii** [comment]
:+1: Click looks like a nice alternative

**Comment by sxooler** [comment]
@gar1t: Is there any update on this topic? I know there is this example https://github.com/guildai/examples/tree/master/flags-dest, which does not auto-detect flags from scripts that use Click.

**Comment by gar1t** [comment]
Preliminary support for importing flags from Click landed in 887e5438.

Note that the example I have above is wrong - flags must be defined as options, not arguments. The correct example is this:

``` python
import click

@click.command()
@click.option("--msg", default="yop")
def hello(msg):
    print(msg)

hello()
```

**Comment by sxooler** [comment]
Thank you! I would really love to use guild and Click together without defining flags separately in guild.yml and Click options.

**Comment by sxooler** [comment]
@gar1t: I have just tried the [example ](https://github.com/guildai/guildai/tree/master/examples/flags) after building guild from source to have the latest update and it doesn't work (_guild: unsupported flags dest: args:click_). I know it's just preliminary support but if it was supposed to work for this short example, it does not (at least for me).

**Comment by gar1t** [comment]
Confirm you have the latest from master.

Also, confirm you're using the built-from-source version of Guild? You can check by running:

    $ guild check

Confirm that `guild_install_location` refers to the local Guild AI source repo.

Also, check that the egg-info is being built from the latest by checking this file:

    $GUILD_REPO/guildai.egg-info/entry_points.txt

This file must contain the following section:

```
[guild.python.flags]
click = guild.plugins.click_flags:ClickFlags
```

If it doesn't, be sure to run `python setup.py build` where `python` is whichever Python version you want to build for.

**Comment by sxooler** [comment]
It was my fault. I did all installation steps correctly but didn't check the path location as `guild` was still pointing to my previous version. Everything works fine now. Thank you for your great effort (on the whole super guild project)!

**Comment by gar1t** [comment]
You're very welcome! Thanks a lot for taking the time and effort to check this!

**Comment by gar1t** [comment]
The fix is now generally available in 0.7.0.
